src/Dom/ResourceHintManager.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace AmpProject\Dom;
+
+use AmpProject\Attribute;
+use AmpProject\RequestDestination;
+use AmpProject\Tag;
+use DOMNode;
+
+/**
+ * Resource hint manager class that is used to manage the <link> tags within a document's <head> to give the browser
+ * hints about how to prioritize resources.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class ResourceHintManager
+{
+
+    /**
+     * Document to manage the resource hints for.
+     *
+     * @var Document
+     */
+    private $document;
+
+    /**
+     * Reference node to attach the resource hint to.
+     *
+     * @var DOMNode|null
+     */
+    private $referenceNode;
+
+    /**
+     * ResourceHintManager constructor.
+     *
+     * @param Document $document
+     */
+    public function __construct(Document $document)
+    {
+        $this->document = $document;
+    }
+
+    /**
+     * Add a preconnect resource hint.
+     *
+     * @param string $href        URL to link to.
+     * @param bool   $crossorigin Whether the link is crossorigin.
```

Comment:
The `$crossorigin` isn't actually a bool, is it? It can have the values `anonymous` or `use-credentials`. If empty then it is the same as `anonymous`: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin